apps/api/src/controllers/user.controller.ts

```diff
@@ -362,56 +303,19 @@ export function makeUserController(services: ServiceRegistry) {
         }
 
         // Prepare update data with only allowed fields
-        const updateData: {
-          id: string;
-          name?: string;
-          description?: string;
-          imageUrl?: string;
-          email?: string;
-          metadata?: Record<string, unknown>;
-        } = {
+        const updateData = {
           id: agentId,
+          name: name ?? agent.name,
+          description,
+          imageUrl,
+          email,
+          metadata,
         };
 
-        if (name !== undefined) {
```

Comment:
example of where we can just defer to zod instead of parsing all of these by hand